src/utils.h

```diff
@@ -0,0 +1,36 @@
+#ifndef UTILS_H
+#define UTILS_H
+
+#define R_NO_REMAP
+
+#include <R.h>
+#include <Rinternals.h>
+
+#define BEGIN_CPP try {
+
+#define END_CPP                                                                \
```

Comment:
I'll add a comment in the code as well, but basically C++ is unsafe to use from R, and if any C++ errors are not caught it will crash the R session, rather than throw an error in R. This is our standard way of putting try-catch guards around C++ code